cmd/kubeadm/app/node/bootstrap.go

```diff
@@ -0,0 +1,138 @@
+/*
+Copyright 2016 The Kubernetes Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    http://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+*/
+
+package node
+
+import (
+	"fmt"
+	"os"
+	"time"
+
+	"github.com/golang/glog"
+
+	kubeadmapi "k8s.io/kubernetes/cmd/kubeadm/app/api"
+	kubeadmutil "k8s.io/kubernetes/cmd/kubeadm/app/util"
+	"k8s.io/kubernetes/pkg/apis/certificates"
+	clientset "k8s.io/kubernetes/pkg/client/clientset_generated/internalclientset"
```

Comment:
`kubeadm` currently calls [this function](https://github.com/kubernetes/kubernetes/blob/master/pkg/kubelet/util/csr/csr.go#L37) for the CSR, which expects an unversioned cert client in its signature and is also currently used by the `kubelet` as well. If I use a versioned client, I cannot directly pass it as first argument to the above function, as it does not implement the interface. What is the correct pattern for dealing with this? Is there some magic way to cast it? I am not 100% familiar with the hierarchy here ...